vllm/entrypoints/openai/cli_args.py

```diff
@@ -82,220 +83,269 @@ def __call__(
+    """If specified, API server will add X-Request-Id header to responses. 
+    Caution: this hurts performance at high QPS."""
+    enable_auto_tool_choice: bool = False
+    """Enable auto tool choice for supported models. Use ``--tool-call-parser`` 
+    to specify which parser to use."""
+    tool_call_parser: Optional[str] = None
+    """Select the tool call parser depending on the model that you're using. 
+    This is used to parse the model-generated tool call into OpenAI API format. 
+    Required for ``--enable-auto-tool-choice``."""
+    tool_parser_plugin: str = ""
+    """Special the tool parser plugin write to parse the model-generated tool 
+    into OpenAI API format, the name register in this plugin can be used in 
+    ``--tool-call-parser``."""
+    log_config_file: Optional[str] = envs.VLLM_LOGGING_CONFIG_PATH
+    """Path to logging config JSON file for both vllm and uvicorn"""
+    max_log_len: Optional[int] = None
+    """Max number of prompt characters or prompt ID numbers being printed in 
+    log. The default of None means unlimited."""
+    disable_fastapi_docs: bool = False
+    """Disable FastAPI's OpenAPI schema, Swagger UI, and ReDoc endpoint."""
+    enable_prompt_tokens_details: bool = False
+    """If set to True, enable prompt_tokens_details in usage."""
+    enable_server_load_tracking: bool = False
+    """If set to True, enable tracking server_load_metrics in the app state."""
+    enable_force_include_usage: bool = False
+    """If set to True, including usage on every request."""
+    expand_tools_even_if_tool_choice_none: bool = False
+    """Include tool definitions in prompts even when ``tool_choice='none'``.
+
+    This is a transitional option that will be removed in v0.10.0. In
+    v0.10.0, tool definitions will always be included regardless of
+    ``tool_choice`` setting. Use this flag to test the upcoming behavior
+    before the breaking change."""
+
+    def __post_init__(self):
+        # Set default values for list fields that should not be None
+        if self.allowed_origins is None:
+            self.allowed_origins = ["*"]
+        if self.allowed_methods is None:
+            self.allowed_methods = ["*"]
+        if self.allowed_headers is None:
+            self.allowed_headers = ["*"]
+        if self.middleware is None:
+            self.middleware = []
+
+    @staticmethod
+    def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
+        from vllm.engine.arg_utils import get_kwargs
+
+        frontend_kwargs = get_kwargs(FrontendArgs)
+        frontend_group = parser.add_argument_group(
+            title="Frontend",
+            description=FrontendArgs.__doc__,
+        )
+
+        frontend_group.add_argument("--host", **frontend_kwargs["host"])
+        frontend_group.add_argument("--port", **frontend_kwargs["port"])
+        frontend_group.add_argument("--uvicorn-log-level",
+                                    **frontend_kwargs["uvicorn_log_level"])
```

Comment:
Since this is a brand new class and there is a 1:1 mapping of arg name to dataclass field name, we could reduce duplication further by doing somethin like:

```python
for field_name, field_kwargs in frontend_kwargs.items():
    argument_name = f"--{field_name.replace('_', '-')}"
    frontend_group.add_argument(argument_name, **field_kwargs)
```

For special cases, we can modify the values of `frontend_kwargs` before the loop.

Reply:
yep. good suggestion. Done.